creator-node/src/fileManager.js

```diff
@@ -102,23 +104,43 @@ async function saveFileToIPFSFromFS (req, srcPath, fileType, sourceFile, transac
  *  - Write file to disk.
  *  - Add file to local inode if not already.
  */
-async function saveFileForMultihash (req, multihash, expectedStoragePath) {
+async function saveFileForMultihash (req, multihash, expectedStoragePath, gatewaysToTry) {
   // If file already stored on disk, return immediately.
   if (fs.existsSync(expectedStoragePath)) {
     req.logger.info(`File already stored at ${expectedStoragePath} for ${multihash}`)
     return expectedStoragePath
   }
 
+  const storagePath = req.app.get('storagePath')
```

Comment:
@raymondjacobson this was part of your changes with https://github.com/AudiusProject/audius-protocol/pull/617. i branched off your old sync everything pr and i modified that code do these ops first and then write the file later.